cpp/src/barretenberg/stdlib/primitives/field/field.test.cpp

```diff
@@ -120,76 +127,95 @@ template <typename Composer> class stdlib_field : public testing::Test {
      */
     static void test_assert_equal()
     {
-        auto run_test = [](bool constrain, bool true_when_y_val_zero = true) {
-            Composer composer = Composer();
-            field_ct x = witness_ct(&composer, 1);
-            field_ct y = witness_ct(&composer, 0);
-
-            // With no constraints, the proof verification will pass even though
-            // we assert x and y are equal.
-            bool expected_result = true;
-
-            if (constrain) {
-                /* The fact that we have a passing test in both cases that follow tells us
-                 * that the failure in the first case comes from the additive constraint,
-                 * not from a copy constraint. That failure is because the assert_equal
-                 * below says that 'the value of y was always x'--the value 1 is substituted
-                 * for x when evaluating the gate identity.
-                 */
-                if (true_when_y_val_zero) {
-                    // constraint: 0*x + 1*y + 0*0 + 0 == 0
-                    add_triple t{ .a = x.witness_index,
-                                  .b = y.witness_index,
-                                  .c = composer.zero_idx,
-                                  .a_scaling = 0,
-                                  .b_scaling = 1,
-                                  .c_scaling = 0,
-                                  .const_scaling = 0 };
-
-                    composer.create_add_gate(t);
-                    expected_result = false;
+        if constexpr (IsSimulator<Composer>) {
```

Comment:
This test illustrates quirks of the circuit `assert_equal`, so I had to make a (much more straightforward) simulator branch.